addons/stock/report/report_stock_reception.py

```diff
@@ -0,0 +1,283 @@
+                if not source:
+                    continue
+                qty_assigned = min(total_assigned, out_move.product_qty)
+                sources_to_lines[source].append(
+                    self._prepare_report_line(qty_assigned, product_id, out_move, source, is_assigned=True, move_ins=moves_in))
+
+        # dates aren't auto-formatted when printed in report :(
+        sources_to_formatted_scheduled_date = defaultdict(list)
+        for source, dummy in sources_to_lines.items():
+            sources_to_formatted_scheduled_date[source] = self._get_formatted_scheduled_date(source)
+
+        return {
+            'data': data,
+            'doc_ids': docids,
+            'doc_model': 'stock.picking',
+            'sources_to_lines': sources_to_lines,
+            'precision': self.env['decimal.precision'].precision_get('Product Unit of Measure'),
+            'pickings': pickings,
+            'sources_to_formatted_scheduled_date': sources_to_formatted_scheduled_date,
+        }
+
+    def _prepare_report_line(self, quantity, product, move_out, source=False, is_assigned=False, is_qty_assignable=True, move_ins=False):
+        return {
+            'source': source,
+            'product': {
+                'id': product.id,
+                'display_name': product.display_name
+            },
+            'uom': product.uom_id.display_name,
+            'quantity': quantity,
+            'is_qty_assignable': is_qty_assignable,
+            'move_out': move_out,
+            'is_assigned': is_assigned,
+            'move_ins': move_ins and move_ins.ids or False,
+        }
+
+    def _get_formatted_scheduled_date(self, source):
+        """ Unfortunately different source record types have different field names for their "Scheduled Date"
+        Therefore an extendable method is needed.
+        """
+        if source._name == 'stock.picking':
+            return format_date(self.env, source.scheduled_date)
+        return False
+
+    def action_assign(self, move_ids, qtys, in_ids):
+        """ Links picking move(s) to the next step (i.e. move_ids in another picking)
+        """
+        outs = self.env['stock.move'].browse(move_ids)
+        # Split outs with only part of demand assigned to prevent reservation problems later on.
+        # We do this first so we can create their split moves in batch
+        out_to_new_out = OrderedDict()
+        new_move_vals = []
+        for out, qty_to_link in zip(outs, qtys):
+            if float_compare(out.product_qty, qty_to_link, precision_rounding=out.product_id.uom_id.rounding) == 1:
+                new_move_vals += out._split(out.product_qty - qty_to_link)
+                out_to_new_out[out.id] = self.env['stock.move']
+        new_outs = self.env['stock.move'].create(new_move_vals)
+        new_outs._action_confirm(merge=False)
+        i = 0
+        for k, dummy in out_to_new_out.items():
+            out_to_new_out[k] = new_outs[i]
+            i += 1
```

Comment:
You can do
for i, k in emumerate(out_to_new_out.keys()):
    out_to_new_out[k] = new_outs[i]

I will fix up it